app/models/ebean/BaseExecution.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2019 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.ebean;
+
+import io.ebean.annotation.DbJsonB;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import javax.persistence.Column;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.MappedSuperclass;
+
+/**
+ * A execution event for a {@link models.internal.Alert}.
+ * <p>
+ * NOTE: This class is enhanced by Ebean to do things like lazy loading and
+ * resolving relationships between beans. Therefore, including functionality
+ * which serializes the state of the object can be dangerous (e.g. {@code toString},
+ * {@code @Loggable}, etc.).
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+// CHECKSTYLE.OFF: MemberNameCheck
+@MappedSuperclass
+public abstract class BaseExecution<T> {
+    private static final String EXCEPTION_KEY = "exception";
+
+    @Id
+    @Column(name = "scheduled")
+    protected Instant scheduled;
+    @Column(name = "state")
+    @Enumerated(value = EnumType.STRING)
+    private State state;
+    @Nullable
+    @Column(name = "started_at")
+    private Instant started_at;
+    @Nullable
+    @Column(name = "completed_at")
+    private Instant completed_at;
+    @Nullable
+    @DbJsonB
+    @Column(name = "error")
+    private Map<String, String> error;
+
+    /**
+     * Get the corresponding Job ID for this execution.
+     *
+     * @return The job ID
+     */
+    public abstract UUID getJobId();
+
+    /**
+     * Set the corresponding Job ID for this execution.
+     *
+     * @param jobId The job ID
+     */
+    public abstract void setJobId(UUID jobId);
+
+    /**
+     * Get the result of this execution, if any.
+     *
+     * @return The result, or null if this execution has not completed.
+     */
+    public abstract @Nullable T getResult();
```

Comment:
Perhaps I'll leave a comment in the code itself, but this needed to be abstract so that the subclass is forced to declare the concrete type as a member. If we instead declared it as a `T` in this class, Jackson wouldn't be able to deserialize it.